public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -180,6 +180,16 @@ export class DashboardModel implements TimeModel {
       }
     }
 
+    // Explicit handling of graph -> time series migration
+    if (options?.autoMigrateOldPanels || !config.angularSupportEnabled || config.featureToggles.autoMigrateGraphPanel) {
+      for (const p of this.panelIterator()) {
+        if (p.type === 'graph') {
+          p.autoMigrateFrom = p.type;
+          p.type = 'timeseries';
```

Comment:
dont love use of magic strings here... not sure though if there is a const somewhere we can draw them from or 🤷‍♂️ seems like they were just [hardcoded before](https://github.com/grafana/grafana/pull/79369/files#diff-cb598ee59be9461b05f4b7529ce1940bf730769842fcecaed4fb8e79961a1bdeL132) so maybe this is ok

Reply:
also not ideal to loop through panels twice here, will reconsider approach when creating explicit feature toggles for the rest of the deprecated panels (can just do one pass through panels and do a switch case or something)

Reply:
on looking deeper into this and failing tests, it seems approach of simply removing graph / timeseries key value from PanelModel's `autoMigrateAngular` dictionary will not work as it is used in other areas of the code - will look at a different approach to account for this

Reply:
> also not ideal to loop through panels twice here

While not ideal, at least in this case doesn't seem like a huge deal as this presumably shouldn't be run all the time.